signac/contrib/migration/v0_to_v1.py

```diff
@@ -6,8 +6,30 @@
 This migration is a null-migration that serves as a template
 for future migrations and testing purposes.
 """
+import os
 
+from signac.common import configobj
 
-def migrate_v0_to_v1(project):
+# A minimal v1 config.
+_cfg = """
+schema_version = string()
+project = string()
+workspace_dir = string(default='workspace')
+"""
+
+
+def _load_config_v1(root_directory):
+    cfg = configobj.ConfigObj(
+        os.path.join(root_directory, "signac.rc"), configspec=_cfg.split("\n")
+    )
+    validator = configobj.validate.Validator()
+    if not cfg.validate(validator):
+        raise RuntimeError(
+            "This project's config file is not compatible with signac's v1 schema."
+        )
```

Comment:
I believe that additional keys will throw here (see https://github.com/glotzerlab/signac/blob/a53a098e8366a785f4572e0872a5788c20c00665/signac/common/config.py#L103-L109). Is that intended?

Reply:
This is a good question that I meant to ask about and forgot!

Reply:
`ConfigObj.validate` will not throw when provided extra keys. It only validates keys that are actually present in the spec. For example:
```
In [1]: >>> from signac.common import configobj

In [2]: >>> _cfg = """
   ...:    ...: schema_version = string()
   ...:    ...: project = string()
   ...:    ...: workspace_dir = string(default='workspace')
   ...:    ...: """

In [3]: >>> cfg = configobj.ConfigObj(configspec=_cfg.split("\n"))

In [4]: >>> validator = configobj.validate.Validator()

In [5]: >>> cfg.validate(validator)
Out[5]: {'schema_version': False, 'project': False, 'workspace_dir': True}

In [6]: >>> cfg.update({'schema_version': '1', 'project': 'foo'})

In [7]: >>> cfg.validate(validator)
Out[7]: True

In [8]: >>> cfg.update({'foo': 'bar'})

In [9]: >>> cfg.validate(validator)  # This returns true despite the extraneous key.
Out[9]: True
```
However, there is a bug here in that I'm checking the truthiness of the validate output rather than checking that it's exactly `True`. I just fixed that.